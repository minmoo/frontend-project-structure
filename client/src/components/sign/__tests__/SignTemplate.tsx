import React from 'react';
import { render } from '@testing-library/react';
import SignTemplate from '../SignTemplate';

describe('<SignTemplate/>', () => {
  //snapshot test
  test('matches snapshot', () => {
    const utils = render(<SignTemplate children title="타이틀" />);
    expect(utils.container).toMatchSnapshot();
  });

  //props
  test('shows the props correctly', () => {
    const utils = render(<SignTemplate children title="타이틀" />);
    utils.getByText('타이틀');
  });
});
