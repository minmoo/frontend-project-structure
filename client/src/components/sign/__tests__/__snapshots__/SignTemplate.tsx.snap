// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignTemplate/> matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-main-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation6 MuiPaper-rounded"
    >
      <div
        class="makeStyles-input-3"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-6 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          타이틀
        </h1>
      </div>
      <div
        class="makeStyles-side-4"
      >
        <h2
          class="MuiTypography-root makeStyles-sideTitle-5 MuiTypography-h2"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 666ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          Welcome
          <br />
           to the
          <br />
           Back Office
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-main-1 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation6 MuiPaper-rounded"
    >
      <div
        class="makeStyles-input-3"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-6 MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5"
        >
          타이틀
        </h1>
      </div>
      <div
        class="makeStyles-side-4"
      >
        <h2
          class="MuiTypography-root makeStyles-sideTitle-5 MuiTypography-h2"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 666ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          Welcome
          <br />
           to the
          <br />
           Back Office
        </h2>
      </div>
    </div>
  </div>
</div>
`;
